common/script/index.coffee

```diff
@@ -72,13 +72,50 @@ api.daysSince = (yesterday, options = {}) ->
   Math.abs api.startOfDay(_.defaults {now:yesterday}, o).diff(api.startOfDay(_.defaults {now:o.now}, o), 'days')
 
 ###
-  Should the user do this taks on this date, given the task's repeat options and user.preferences.dayStart?
+  Should the user do this task on this date, given the task's repeat options and user.preferences.dayStart?
 ###
-api.shouldDo = (day, repeat, options={}) ->
-  return false unless repeat
+api.shouldDo = (day, dailyTask, options = {}) ->
+  return false unless dailyTask.type == 'daily' && dailyTask.repeat
+  if !dailyTask.startDate
+    # TODO: Unexpected code path reached. Log a warning.
+    dailyTask.startDate = moment().toDate();
+  if dailyTask.startDate instanceof String
+    #TODO: Unexpected code path reached. Log a warning.
+    dailyTask.startDate = moment(dailyTask.startDate).toDate();
```

Comment:
No, you're right. I forgot that this code is just on the client, so logging makes no sense.